### PR TITLE
(docs) Revise the agent compatibility page

### PR DESCRIPTION
**Pull request by nfagerlund**

### CODE REVIEW

**Comment by jeffmccune** [comment]
:+1:

File: documentation/compatibility_with_puppet_agent.markdown [hunk 1]
```diff
@@ -1,106 +1,100 @@
...
+
+## Preparing Puppet 3 Nodes for Puppet Server 2
+
+Backwards compatibility is enabled by default, but you should make sure your Puppet code is ready for Puppet 4 before pointing old agent nodes at a new Puppet server. You may also need to edit auth.conf.
+
+Before migrating Puppet 3 nodes to Puppet Server 2, do all of the following:
+
+- Update Puppet to the most recent 3.x version on your agent nodes and existing Puppet master.
+- Set `stringify_facts = false` on all agent nodes, and fix your code if necessary.
+- Make sure you're already running with the future parser enabled, and fix your code if necessary.
```
Comment (dankreek) [inline documentation/compatibility_with_puppet_agent.markdown]: Does this require any further documentation? Or maybe a link to some?

Comment (dankreek) [inline documentation/compatibility_with_puppet_agent.markdown]: And I spoke too soon.. loos like it's directly below. Hmm maybe just add "see instructions below" or something?

**Comment by camlow325** [comment]
:+1: will let @dankreek merge when he's done reviewing if he's good with it.

**Comment by dankreek** [comment]
All-in-all looks fine to me. What do you think of my comment, @nfagerlund? (I'm not the docs expert so I'm deferring to you on that.)

**Comment by nfagerlund** [comment]
Eh, sure, wouldn't hurt.

**Comment by nfagerlund** [comment]
@dankreek Should be good to go now!

**Comment by dankreek** [comment]
:+1: looks good!

**Pull request merged**